src/library/features/crates/cratetreemodel.h

```diff
@@ -0,0 +1,43 @@
+#ifndef CRATE_TREE_MODEL_H
+#define CRATE_TREE_MODEL_H
+
+#include <QList>
+#include <QMap>
+
+#include "library/treeitemmodel.h"
+#include "library/features/crates/cratehierarchy.h"
+#include "library/crate/crate.h"
+
+// This class represents the crate tree for the nested crates feature
+
+class CrateTreeModel : public TreeItemModel {
+  public:
+    CrateTreeModel(LibraryFeature* pFeature,
+                   TrackCollection* pTrackCollection,
+                   CrateHierarchy* pCrateHierarchy);
+    //virtual ~CrateTreeModel();
+
+    // QVariant data(const QModelIndex& index, int role) const override;
+    // bool setData(const QModelIndex& index, const QVariant& value, int role) override;
+
+  public slots:
+    void reloadTree() override;
+
+  protected:
+
+  private:
+    bool findParentChildFromPath(Crate& parent,
```

Comment:
This function could have a better name. At first glance it seems like it is finding a "ParentChild" from the given path and the bool return type doesn't make sense with that idea. How about "findParentAndChildFromPath"? It's a long name, but it is clear what the function does. There should be a comment here in the header file explaining what the returned bool indicates.